tests/functional/test_no_color.py

```diff
@@ -0,0 +1,61 @@
+"""
+Test specific for the --no-color option
+"""
+import os
+import platform
+import subprocess as sp
+import sys
+
+import pytest
+
+
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="does not run on windows")
+def test_no_color(script):
+
+    """
+    Test uninstalling an existing package - should out put red error
+
+    We must use subprocess with the script command, since redirection
```

Comment:
Are you sure?

`pip uninstall NoSuchPackage | cat` still gives me coloured output.

Reply:
Yes I am sure. Try the following sequences to verify:
```
[pr-4739]  $ pip uninstall NoSuchPackage | cat
Cannot uninstall requirement NoSuchPackage, not installed
# this would be red
[pr-4739]  $ pip uninstall NoSuchPackage > file.txt
Cannot uninstall requirement NoSuchPackage, not installed
# this would be red too, we only redirected stdout, not stderr
[pr-4739]  $ pip uninstall NoSuchPackage 2> file.txt
# nothing, everything is in that file.
[pr-4739]  $ cat file.txt 
Cannot uninstall requirement NoSuchPackage, not installed
# text is there, but it is not red anymore.
```

Reply:
Oh right. My bad. It's printed on stderr. :/

```sh
pip uninstall NoSuchPackage 2>&1 | cat 
```

Reply:
@pradyunsg No problem. I implemented the simplifications you asked for. Now travis has to do it's thing.